site/content/en/docs/user/metrics-configuration.md

```diff
@@ -0,0 +1,143 @@
+
+
+The `metrics` field are used to customize the return body of the installed metrics endpoint.
+`metrics` is a list of specific configuration items, with each corresponding to a Prometheus style metric:
+* `name` defines the metric name.
+* `labels` defines the metric labels, with each item corresponding to a specific metric label.
+  - `name` is a const string that provides the label name.
+  - `value` is represented as a CEL expression that dynamically determines the label value.
+    For example: you can use `pod.metadata.name` to reference the pod name as the label value.
+* `help` defines the help string of a metric.
+* `kind` defines the type of the metric: `counter`, `guage` or `histogram`.
+* `dimension` defines where the data comes from. It could be `node`, `pod`, or `container`.
+* `value` is a CEL expression that defines the metric value if `kind` is `counter` or `guage`.
+  Please refer to [built-in CEL extension functions] for an exhausted list that you can use to simulate the metric value.
+* `buckets` is exclusively for customizing the data of the metric of kind `histogram`.
+  - `le` has the same meaning as the one of Prometheus histogram bucket.
+  - `value` is a CEL expression that provides the value of the bucket.
+  - `hidden` indicates whether to hide the bucket. The bucket would be ignored if set to `true`.
+
+Please refer to [Metrics for kubelet's "metrics/resource" endpoint] for a detailed example.
+
+
+## built-in CEL extension functions
+
+As `kwok` doesn't actually run containers, to simulate the resource usage data, `kwok` provides the following CEL
+extension functions you can use in a CEL expression to help calculate the simulation data flexibly and dynamically:
+* `Now()`: yields the current timestamp.
+* `Rand()`: yields a random `float64` value.
+* `SinceSecond()` yields the seconds elapsed since a resource (pod or node) was created.
+  For example: `pod.SinceSecond()`, `node.SinceSecond()`.
+* `UnixSecond()` yields the Unix time of a given time of type `time.Time`.
+  For example: , `UnixSecond(Now())`, `UnixSecond(node.metadata.creationTimestamp)`.
+* `Quantity()` yields a float64 value of a given Quantity value. For example: `Quantity("100m")`, `Quantity("10Mi")`.
+* `Usage()` yields the current instantaneous resource usage with the simulation data in [ResourceUsage (ClusterResourceUsage)].
+  For example: `Usage(pod, "memory")`, `Usage(node, "memory")`, `Usage(pod, "memory", container.name)` yield the
+  current working set of a resource (pod, node or container) in bytes.
+* `CumulativeUsage()` yield the cumulative resource usage in seconds with the simulation data given in [ResourceUsage (ClusterResourceUsage)].
+  For example: `CumulativeUsage(pod, "cpu")`, `CumulativeUsage(node, "cpu")`, `CumulativeUsage(pod, "cpu", container.name)`
+  yield a cumulative cpu time consumed by a resource (pod, node or container) in core-seconds.
```

Comment:
```suggestion
* `Now()`: returns the current time, of type `time.Time`
* `Rand()`: returns a random `float64` value.
* `SinceSecond()` returns the seconds elapsed since a resource (pod or node) was created.
  For example: `pod.SinceSecond()`, `node.SinceSecond()`.
* `UnixSecond()` returns the Unix time of a given time of type `time.Time`.
  For example: , `UnixSecond(Now())`, `UnixSecond(node.metadata.creationTimestamp)`.
* `Quantity()` returns a float64 value of a given quantity value. For example: `Quantity("100m")`, `Quantity("10Mi")`.
* `Usage()` returns the current instantaneous resource usage with the simulation data in [ResourceUsage (ClusterResourceUsage)].
  For example: `Usage(pod, "memory")`, `Usage(node, "memory")`, `Usage(pod, "memory", container.name)` yield the
  current working set of a resource (pod, node or container) in bytes.
* `CumulativeUsage()` returns the cumulative resource usage with the simulation data given in [ResourceUsage (ClusterResourceUsage)].
  For example: `CumulativeUsage(pod, "cpu")`, `CumulativeUsage(node, "cpu")`, `CumulativeUsage(pod, "cpu", container.name)`
  returns a cumulative cpu time consumed by a resource (pod, node or container).
```